Zone5/Transport/URLRequestInterceptor.swift

```diff
@@ -148,20 +149,39 @@ internal class URLRequestInterceptor: URLProtocol {
 		
 		// now we are finished with the request meta. Clear it before we continue
 		let request = request.clearMeta()
+		let currentTask: URLSessionTask
 		
 		switch type {
 		case .data:
-			self.currentTask = session.dataTask(with: request, completionHandler: onComplete)
+			currentTask = session.dataTask(with: request, completionHandler: onComplete)
+
 		case .upload:
-			if let url = url {
-				self.currentTask = session.uploadTask(with: request, fromFile: url, completionHandler: onComplete)
-			} else {
+			guard let url = url else {
 				onComplete(data: nil, response: nil, error: Zone5.Error.invalidParameters)
+
+				return
+			}
+
+			currentTask = session.uploadTask(with: request, fromFile: url)
+
+			var completionObserver : NSObjectProtocol? = nil
+			completionObserver = NotificationCenter.default.addObserver(forName: URLRequestDelegate.uploadCompleteNotification, object: currentTask, queue: nil) { [ weak self ] notification in
+				// remove observers.
+				if let completionObserver = completionObserver {
+					NotificationCenter.default.removeObserver(completionObserver, name: URLRequestDelegate.downloadCompleteNotification, object: currentTask)
```

Comment:
I'm guessing this is supposed to be remove uploadCompleteNotification, not downloadCompleteNotification

Reply:
It sure is!